src/java/org/apache/cassandra/io/util/ChecksumWriter.java

```diff
@@ -91,11 +91,38 @@ public void appendDirect(ByteBuffer bb, boolean checksumIncrementalResult)
     }
 
     public void writeFullChecksum(@Nonnull File digestFile)
+    {
+        writeFullChecksum(digestFile, fullChecksum.getValue());
+    }
+
+    /**
+     * Repopulate checksum value from given source file and write to digest file
+     */
+    public void repopulateChecksum(@Nonnull File source, @Nonnull File digestFile)
+    {
+        try (RandomAccessReader in = RandomAccessReader.open(source))
+        {
+            CRC32 checksum = new CRC32();
+            CheckedInputStream checkedInputStream = new CheckedInputStream(in, checksum);
+
+            byte[] chunk = new byte[64 * 1024];
+            while( checkedInputStream.read(chunk) > 0 ) { }
```

Comment:
Usually the contract for read() in InputStreams in Java is that the end is -1.
It is allowed to have a read () returning zero.
Maybe this is not a problem here?